### ISSUE TITLE
Responsive feature using unslick wipes out containing divs

### ISSUE THREAD

**Issue by yenly** [akiran/react-slick#1172]
Example code here:
https://codesandbox.io/s/j1l1nn0695
https://z613xn9pk3.codesandbox.io/

Please look at the above example and change the browser size to see this critical bug.

Version 22.3 didn't have this issue. Latest version replaces slide content with empty div. 
<img width="1550" alt="screen shot 2018-04-05 at 11 29 40 am" src="https://user-images.githubusercontent.com/6759658/38384878-286952fc-38c5-11e8-8433-b1e7aa6a050d.png">
As you can see in the above screen capture, once the browser is in the setting for responsive unslick, it hits the  zone where everything in <div class='slick-slider> becomes an empty div.

**Comment by danilobuerger** [comment]
Can confirm, this worked for me in 0.22.3 and is broken afterwards.

**Comment by misstricky** [comment]
Any word as to when this fix is going to be released? Thanks!

**Comment by ryanboswell** [comment]
I don't usually like to "bump" things, but any hope for a release with this fix sometime soon?

**Comment by johnrom** [comment]
I can confirm that the fix in 56b124da9509370a33682677f205c2092f59b648 works

**Comment by AndyArcherKG** [comment]
Is there a fix of this live now? 
i'm checking `0.23.1` and when i "unslick" it wipes all my content.

**Comment by misstricky** [comment]
Yes I think unslick is fixed, but, to work around what I think you're experiencing, I just went ahead and skipped all unslick behavior and put resizes listeners in my component, and then only initiate the slideshow when I need it, something like...

```
 getSettings() {
        return {
            slidesToShow: 1,
            slidesToScroll: 1,
            centerMode: true,
            centerPadding: this.props.centerPadding,
            speed: 300,
            arrows: false,
            adaptiveHeight: false
        };
    }

    render() {
        const settings = this.getSettings();
        let filterNull = React.Children.toArray(this.props.children).filter(
            (item) => item !== null
        );

        if (
            this.state.windowWidth <= this.props.startBreakpoint &&
            filterNull &&
            filterNull.length > 1
        ) {
            return <Slider {...settings}>{this.props.children}</Slider>;
        } else {
            return <div>{this.props.children}</div>;
        }
    }
}
```

This is probably not the most helpful example, but, the react-slick `Slider` will only render if the window is less than the breakpoint prop here, otherwise do not render slideshow. Just not using unslick anymore.

**Comment by AndyArcherKG** [comment]
@misstricky when you say theres a fix live - is there an npm version for it?

**Comment by misstricky** [comment]
@AndyArcherKG I don't work on this project (use it a lot though!) -- but I saw that unslick being fixed was listed here, this release: https://github.com/akiran/react-slick/releases/tag/0.19.0

-- So I think it works but I'm not sure how its really intended to work, so I just work around it. Hope that helps! Might want to ask others who work on this project!

**Comment by AndyArcherKG** [comment]
@misstricky yeah thats fair enough i appreicate your example. I've grabbed version `0.19.0`and this now seems to not destroy it when it unslicks. So definitly confirms theres a bug in `0.23.1`

**Comment by t0mclaudio** [comment]
> How about using `settings: { unslick: true }`? it works at `0.23.2` for me

This worked for me in version 0.23.2. Took me half a day looking for solution

**Comment by MrAvantiC** [comment]
This is still occuring on version 0.23.2.
Unfortunately `settings: { unslick: true }` does not work either...

**Comment by AllanPooley** [comment]
@mskims @akiran @laveesingh this is still an issue

Using settings:

```
const sliderSettings = {
    dots: false,
    arrows: true,
    infinite: true,
    autoplay: false,
    autoplaySpeed: 4000,
    speed: 800,
    slidesToShow: 1,
    slidesToScroll: 1,
    responsive: [
      {
        breakpoint: 1080
      },
      {
        breakpoint: 99999,
        settings: 'unslick',
        unslick: true
      }
    ]
};
```

I've had to revert to **0.22.3** to get the intended functionality

```
yarn remove react-slick
yarn add react-slick@0.22.3
```

**Comment by haniotis** [comment]
Still an issue

**Comment by AllanPooley** [comment]
Still an issue as of July 2019

A workaround if you need functionality from more recent versions is to use conditional rendering:

```
import { Carousel, List } from '..';

const isClient = typeof window !== 'undefined';
const MOBILE_BREAKPOINT = 800;

class DynamicList extends Component {
  state = {
    viewportWidth: 0,
  };

  componentDidMount() {
    if (isClient) {
      this.updateWindowDimensions();
      window.addEventListener('resize', this.updateWindowDimensions);
    }
  }

  componentWillUnmount() {
    if (isClient) window.removeEventListener('resize', this.updateWindowDimensions);
  }

  updateWindowDimensions = () => {
    this.setState({ viewportWidth: window.innerWidth });
  }

  render () {
    const { items } = this.props;
    const isMobile = Boolean(viewportWidth <= MOBILE_BREAKPOINT);
    return (
      <section className="dynamic-list">
        { isMobile ? (
          <Carousel slides={items} />
        ) : (
          <List items={items} />
        )}
      </section>
    );
  }
};
```

**Comment by AlejoSalvo95** [comment]
I am having exactly the same issue. I had to revert to react-slick@0.22.3 as @AllanPooley said to make it work

**Comment by cembakca** [comment]
I am having exactly the same issue. But as a workaround I did something like this;

`import React from 'react';
import Slider from 'react-slick';

// find better class name :D
export default class UnslickManagement extends React.Component {
  constructor() {
    super();
    this.state = {
      slideVersion: false
    }
  }

  componentDidMount() {
    this.breakpointChecker();
  }

  breakpointChecker = () => {
    // to: https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia
    //Polyfills for old IE support  
    // matchMedia support from media-match (https://github.com/weblinc/media-match)
    const breakpoint = window.matchMedia('(min-width: 770px)');
    if (breakpoint.matches) {
	    this.setState({ slideVersion: false });
    } else {
      this.setState({ slideVersion: true });
    }
  }

  renderSlideVersion = () => {
    return (
      <Slider {...this.props.settings} className={this.props.className}>
         {this.props.children}
      </Slider>
    );
  }

  renderNormalVersion = () => (
    <div className={this.props.className}>
      {this.props.children}
    </div>
  );
  
  render() {
    return this.state.slideVersion ? this.renderSlideVersion() : this.renderNormalVersion();
  }
}`

hopefully it will recover soon

**Comment by tusharshuklaa** [comment]
Any hopes for this to be fixed in March 2020??

**Comment by CuongCycle** [comment]
Hello guys,
I got this problem too.
I also revert to 0.22.3 but got another warning about `componentWillMount` and  `componentWillReceiveProps`.
So I have to use 0.25.2 and fix the unslick by using this:

settings: {
          unslick: true,
          slidesToShow: 1,
          slidesToScroll: 1,
        },

Everything is fine for now. 
Hope this help.

**Comment by akiran** [comment]
Fixed in react-slick@0.26.0

**Comment by kheangseng** [comment]
I fixed it by unslick in all the breaking points (if the slidesToShow is greater than the length of array of elements/contents.

`const getSettings = (arrContents: Content[]) =>{
    const arrLen = arrContents?arrContents.length : 0

    const settings = {     
      dots: true,
      arrows: true,      
      autoplay: false,
      infinite: true,      
      slidesToShow: 6,
      slidesPerRow: 1,          
      speed: 1000,      
      pauseOnHover: true,      
      responsive: [
        {
          breakpoint: 10000,  //to reach all the max screen size
          settings: arrLen <= 6 
            ? {
                unslick: true,
                slidesToShow: arrLen,
              }
            : {            
                slidesToShow: 6,
                slidesToScroll: 6,
                infinite: true,
                dots: true
              }
        },
        {
          breakpoint: 1200,
          settings: arrLen <= 6 
            ? {
                unslick: true,
                slidesToShow: arrLen,
              }
            : {            
                slidesToShow: 6,
                slidesToScroll: 6,
                infinite: true,
                dots: true
              }
        },
        {
          breakpoint: 1024,
          settings: arrLen <= 4 
            ? {
                unslick: true,
                slidesToShow: arrLen,
              }
            : {            
                slidesToShow: 4,
                slidesToScroll: 4,
                infinite: true,
                dots: true
              }
        },
        {
          breakpoint: 768,
          settings: arrLen <= 3 
            ? {
                unslick: true,
                slidesToShow: arrLen,
              }
            : {            
                slidesToShow: 3,
                slidesToScroll: 3,
                // initialSlide: 1,
                infinite: true,
                dots: true
              }            
        },
        {
          breakpoint: 600,
          settings: arrLen <= 2 
            ? {
                unslick: true,
                slidesToShow: arrLen,
              }
            : {            
                slidesToShow: 2,
                slidesToScroll: 1,
                // initialSlide: 1
              }
        },
        {
          breakpoint: 480,
          settings: arrLen <= 1 
            ? {
                unslick: true,
                slidesToShow: arrLen,
              }
            : {            
                slidesToShow: 1,
                slidesToScroll: 1,
              }
        }
      ]
    }
    return settings
  }`

in render, 
`<Slider {...getSettings(recentContents)}>
 ...
</Slider>`
